### PR TITLE
:seedling: Change image-building pipeline to a matrix

### PR DESCRIPTION
**Pull request by mquhuy**
This is to split ubuntu and centos image building into separate VMs and have proper indication for each step results in Jenkins UI.

### CODE REVIEW

File: jenkins/jobs/image_building.pipeline [hunk 1]
```diff
@@ -15,61 +14,88 @@ pipeline {
...
-          usernamePassword(credentialsId: 'metal3ci_city_cloud_openstack_credentials', usernameVariable: 'OPENSTACK_USERNAME_CLEURA', passwordVariable: 'OPENSTACK_PASSWORD_CLEURA'),
-          usernamePassword(credentialsId: 'infra-nordix-artifactory-api-key', usernameVariable: 'RT_USER', passwordVariable: 'RT_TOKEN')
-        ]) {
-          catchError([stageResult: 'FAILURE', message: "Failed to build Centos CI image"]) {
-            sh "./jenkins/image_building/build-image.sh"
+        environment {
+          IMAGE_OS = "${IMAGE_OS}"
+        }
+        stages {
+          stage("Checkout CI Repo") {
```
Comment (Sunnatillo) [inline jenkins/jobs/image_building.pipeline:30]: Can we merge stage("Checkout CI Repo") and stage("Build CI image"). Checking out does not necessarily need to have its own stage. We are having too many granular steps on the view.  

image

File: jenkins/image_building/upload-node-image.sh [hunk 2]
```diff
@@ -68,3 +68,7 @@ upload_node_image() {
         echo "${MAPFILE[i]} has been deleted!"
     done
 }
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
```
Comment (tuminoid) [inline jenkins/image_building/upload-node-image.sh:72]: This is unintuitive for readers, put a comment there what it does and when it is triggered

File: jenkins/jobs/image_building.pipeline [hunk 3]
```diff
@@ -1,10 +1,9 @@
 ci_git_credential_id = "metal3-jenkins-github-username-token"
-ci_git_branch = "main"
-ci_git_url = "https://github.com/metal3-io/project-infra.git"
+ci_git_branch = "mquhuy/image-build-matrix"
```
Comment (tuminoid) [inline jenkins/jobs/image_building.pipeline:2]: Not mergerable with this, so
/hold

File: jenkins/jobs/image_building.pipeline (hunk 1 above)
Comment (mquhuy) [reply jenkins/jobs/image_building.pipeline:30]: I have no big issue, but I tend towards disagreeing with that. We've most likely always put the checkout as a step in all of our jobs, I'd prefer to keep it that way for consistency.

File: jenkins/image_building/upload-node-image.sh (hunk 2 above)
Comment (mquhuy) [reply jenkins/image_building/upload-node-image.sh:72]: Thanks. I thought that was a common practice, but I could be wrong.

**Review by tuminoid — COMMENTED** [review]
/approve
/cc @Sunnatillo @kashifest

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Pull request merged**